File: yangxj96-bean/src/main/java/io/github/yangxj96/enums/system/DictionariesType.java
package io.github.yangxj96.enums.system;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.Getter;

/**
 * 字典类型
 */
@Getter
public enum DictionariesType {

    GROUP(1, "字典组"),
    ITEM(2, "字典项");

    @EnumValue
    private final int code;

    @JsonValue
    private final String name;

    DictionariesType(int code, String name) {
        this.code = code;
        this.name = name;
    }
}

File: yangxj96-bean/src/main/java/com/yangxj96/saas/enums/system/DictionariesType.kt
package com.yangxj96.saas.enums.system

import com.baomidou.mybatisplus.annotation.EnumValue
import com.fasterxml.jackson.annotation.JsonValue

/**
 * 字典类型
 */
enum class DictionariesType(@EnumValue val code: Int, @JsonValue val type: String) {
    GROUP(1, "字典组"),
    ITEM(2, "字典项")
}
